c10/core/InferenceMode.cpp

```diff
@@ -0,0 +1,10 @@
+
+#include <c10/core/InferenceMode.h>
+
+namespace c10 {
+
+bool InferenceMode::is_enabled() {
+  return c10::impl::tls_is_dispatch_key_included(DispatchKey::InplaceOrView);
```

Comment:
This is gonna be hella confusing without a comment, I think. I'd even think about reverting this back to `!` and making `is_dispatch_keyset_included` do the xor, i.e. encapsulate the new technique. But if you want to avoid the extra instruction or whatever, I'd consider at least renaming to `is_dispatch_keyset_included_in_raw_storage` to clarify what the function is actually computing...

Reply:
I actually have a note in InferenceMode.h explaining the expected state of InplaceOrView in InferenceMode which I think can explain it. I'll add a link to note here! :D